### ISSUE TITLE
PTP domain issues

### ISSUE THREAD

**Issue by nm987** [microsoft/W32Time#6]
When following the guide for PTP setup in Windows 10 it seems that W32Time will only sync if PTP domain is 0. I used 2 setups, one is Trimble GM200 grandmaster and another is Tektronix SPG8000 grandmaster + Mellanox SN2010 switch as boundary clock and the end result is the same.  There are no events in "PTP-Operational" when PTP domain is different from 0. 

Here is the output of w32tm /query /status /verbose when PTP domain is 0:
>Leap Indicator: 0(no warning)
Stratum: 3 (secondary reference - syncd by (S)NTP)
Precision: -23 (119.209ns per tick)
Root Delay: 0.0010918s
Root Dispersion: 0.0100020s
ReferenceId: 0x4D505450 (source IP:  77.80.84.80)
Last Successful Sync Time: 10/14/2020 1:48:04 PM
Source: 192.168.200.250
Poll Interval: 10 (1024s)

>Phase Offset: 0.7156279s
ClockRate: 0.0156249s
State Machine: 2 (Sync)
Time Source Flags: 0 (None)
Server Role: 0 (None)
Last Sync Error: 0 (The command completed successfully.)
Time since Last Good Sync Time: 1.9590114s

Here is when PTP domain is different from 0:
>Leap Indicator: 3(not synchronized)
Stratum: 0 (unspecified)
Precision: -23 (119.209ns per tick)
Root Delay: 0.0000000s
Root Dispersion: 0.0000000s
ReferenceId: 0x00000000 (unspecified)
Last Successful Sync Time: unspecified
Source: Local CMOS Clock
Poll Interval: 10 (1024s)

>Phase Offset: 0.0000000s
ClockRate: 0.0156250s
State Machine: 0 (Unset)
Time Source Flags: 0 (None)
Server Role: 0 (None)
Last Sync Error: 1 (The computer did not resync because no time data was available.)
Time since Last Good Sync Time: 19.4881785s``

Events are bein populated only when PTP domain is 0 
![event](https://user-images.githubusercontent.com/72848837/95948386-6e30b300-0e2b-11eb-8357-7a0be4413d22.png)

And lastly, here are the PTP settings of Trimble grandmaster. Using the same settings with PTP domain different from the 0 will not work with W32Time
![trimble](https://user-images.githubusercontent.com/72848837/95948700-0890f680-0e2c-11eb-9279-c615885e53c0.png)

Is there any way to change the PTP domain via registry or similar or there is indeed an issue where only PTP domain 0 is supported?

**Comment by mwtrigg** [comment]
Not that I've seen.  There was a blog post (will edit if I find it) where the poster (MSFT employee) states it was not an option and in the comments didn't understand what the use case would be.  Never mind the fact that it is a part of the RFC.

**Comment by Donis-** [comment]
Yes, unfortunately windows PTP client only syncs with Domain Number 0.
Our use case:
- we're receiving a PTP service that is delivered on a domain number other than 0
- not possible to use windows server 2019 native PTP client.
